core/include/crypto/crypto.h

```diff
@@ -251,6 +251,20 @@ TEE_Result crypto_acipher_sm2_dsa_verify(uint32_t algo,
 					 struct ecc_public_key *key,
 					 const uint8_t *msg, size_t msg_len,
 					 const uint8_t *sig, size_t sig_len);
+TEE_Result crypto_acipher_sm2_kep_derive(uint8_t *out, size_t out_len,
```

Comment:
15 arguments, that has to be some record. I think we need some helper struct to keep track of at least some of the arguments because this function will be very hard to call without making mistakes.

Reply:
I tried using a struct just to see if if would improve things, but actually it is more lines of code and there is only one call site for this function anyway...
I can still push a fixup patch so we can see actual code.